### DISCUSSION TITLE
<Using Quarto in Rstudio?>

### DISCUSSION THREAD

**Discussion by PHUoL** [quarto-dev/quarto-cli#6214] [Q&A]
### Description

My apologies if this questions has been asked somewhere else, but i have not been able to find the answer. Does quarto come pre-installed with Rstudio? I'm using Rstudio version 2023-06.1+524 on MacOS on an M1 running the app in Universal mode. (Ventura). There is instructions on the quarto.org site that still suggests installing quarto. I have found that Quarto works out of box on Windows.  I get the error "state not recoverable" in a pop-up window when I select File -> New File -> Quarto Document... 

I have looked via terminal and have found quarto in /Applications/Rstudio.app/Contents/Resources/app/quarto/bin 

I also have not been able to use Rmarkdown

rmarkdown::find_pandoc()
Error in strsplit(info, "\n")[[1]] : subscript out of bounds
In addition: Warning message:
sh: /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/pandoc: Bad CPU type in executable
In system(paste(shQuote(path), "--version"), intern = TRUE) :
  running command ''/Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/pandoc' --version' had status 126

Pandoc is in /Applications/Rstudio.app/Contents/Resources/app/quarto/bin/tools/

DO I need to run Rstudio  "Run in Rosetta2" to get both Rmarkdown or Quarto to work? I have a working Tex distubution, wiht texlive.  I have read that somewhere that quarto is still not ready for arm64 for mac but a future 1.4 version would be.

**Comment by mcanouil** [comment]
Could you format your post either using the interface or directly markdown such as code block for code?
Also, could you edit the title of your post and remove `<>`?

**Comment by eteitelbaum** [comment]
I had this same question because initially we were being told to update Quarto manually. But according to the [User Guide](https://docs.posit.co/ide/user/ide/guide/documents/quarto-project.html) Quarto is now being updated to the most recent stable version whenever you update RStudio.

**Reply by cderv** [reply]
Updating RStudio will upgrade Quarto, but I believe it will upgrade to latest stable release of Quarto at the time of the RStudio IDE update. 

Quarto could be release in different cycle and not at the same time as an IDE release, so you could still need to update manually if you want to benefit from latest Quarto feature before waiting for a RStudio IDE update

**Comment by cderv** [comment]
> Does quarto come pre-installed with Rstudio?

Yes RStudio IDE does bundle a Quarto binary when there is binary to bundle

> I'm using Rstudio version 2023-06.1+524 on MacOS on an M1 (...) 
> DO I need to run Rstudio "Run in Rosetta2" to get both Rmarkdown or Quarto to work?

This is the specific that requires so more work I think. There are several issue in RStudio repo about this, among them this one: 
* https://github.com/rstudio/rstudio/issues/12791

A warning about this requirement in Desert Sunflower release (2023.09 series). So you should update to latest RStudio IDE available which is now 2023.12 series (https://posit.co/download/rstudio-desktop/)

This comments explains what to do: https://github.com/rstudio/rstudio/issues/12791#issuecomment-1688473620

and indeed Quarto 1.4 that you can get from our download page should solve this by not requiring rosetta as we have a new build 

By using RStudio 2023.06, you are also using an old Quarto version. Know that you can update Quarto anytime on your system. RStudio IDE should pick it up from PATH and not use its bundled version.
